### ISSUE TITLE
reorderGroupBreadthFirst incorrectly orders operations

### ISSUE THREAD

**Issue by rdspring1** [csarofeen/pytorch#403]
## 🐛 Bug

**reorderGroupBreadthFirst** incorrectly orders the operations in batch norm persistent kernel

Current Workaround: Disabled reorderGroupBreadthFirst

## To Reproduce
- Use rds_persistent_kernels_20_9_25 branch
- Uncomment [reorderGroupBreadthFirst](https://github.com/csarofeen/pytorch/blob/20_9_25_devel/torch/csrc/jit/codegen/cuda/lower_loops.cpp#L657) in lower_loops.cpp
- Run `./build/bin/test_jit --gtest_filter='NVFuserTest*FusionSmemDynamicPersistentBatchNorm*'`

**Incorrect Batch Norm Persistent Kernel**
```
__global__ void CUDAGeneratedKernel(Tensor<float, 2> T0, float f4, float f5, float f6, int64_t i7, int64_t i22, Tensor<float, 2> T13) {
  alignas(4) extern __shared__ char array[];
  unsigned offset = 0;
  void* shared_mem = array;
  offset += ((blockDim.x * blockDim.y * blockDim.z) * sizeof(float));

  offset = alignBufferSize(offset,4);
  float* T14 = reinterpret_cast<float*>(array + offset);
  offset += (((ceilDiv(T0.size[1], i22)) * i22) * sizeof(float));

  offset = alignBufferSize(offset,4);
  float* T4 = reinterpret_cast<float*>(array + offset);
  offset += (((ceilDiv(T0.size[1], i22)) * i22) * sizeof(float));

  float T10[1];
  float T9[1];
  float T3[1];
  float T8[1];
  float T2[1];
  float T7[1];

  float T1[1];
  T1[0] = float(0);

  float T6[1];
  T6[0] = float(0);

  float T15[1];
  T15[0] = float(0);

  float T16[1];
  T16[0] = float(0);
  for(size_t i4 = 0; i4 < (ceilDiv(T0.size[1], i22)); ++i4) {
    if ((((i4 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T14[((i4 * blockDim.x) + threadIdx.x)]
         = T0[(blockIdx.x * T0.stride[0]) + (((i4 * i22) + threadIdx.x) * T0.stride[1])];
    }
  }

  for(size_t i10 = 0; i10 < (ceilDiv(T0.size[1], i22)); ++i10) {
    float T5[1];
    if ((((i10 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T5[0]
        = T4[((i10 * blockDim.x) + threadIdx.x)]
        * T4[((i10 * blockDim.x) + threadIdx.x)];
    }

    if ((((i10 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T16[0]
        = T16[0]
        + T5[0];
    }
  }
  __syncthreads();
  for(size_t i19 = 0; i19 < (ceilDiv(T0.size[1], i22)); ++i19) {
    if ((((i19 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T15[0]
        = T15[0]
        + T14[((i19 * blockDim.x) + threadIdx.x)];
    }
  }

  blockReduce<true, false, false>(
    T6[0],
    T16[0],
    [](float &a, float b) { a = a + b; },
    threadIdx,
    blockDim,
    static_cast<float*>(shared_mem),
    1,
    float(0));

  blockReduce<true, false, false>(
    T1[0],
    T15[0],
    [](float &a, float b) { a = a + b; },
    threadIdx,
    blockDim,
    static_cast<float*>(shared_mem),
    1,
    float(0));

  broadcast::blockBroadcast<true, false, false>(
    T7[0],
    T6[0],
    static_cast<float*>(shared_mem));

  broadcast::blockBroadcast<true, false, false>(
    T2[0],
    T1[0],
    static_cast<float*>(shared_mem));

  T8[0]
    = T7[0]
    / i7;

  T3[0]
    = T2[0]
    / i7;

  T9[0]
    = T8[0]
    + f6;

  for(size_t i63 = 0; i63 < (ceilDiv(T0.size[1], i22)); ++i63) {
    if ((((i63 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T4[((i63 * blockDim.x) + threadIdx.x)]
        = T14[((i63 * blockDim.x) + threadIdx.x)]
        - T3[0];
    }
  }

  T10[0]
     = rsqrtf(T9[0]);
  __syncthreads();

  for(size_t i75 = 0; i75 < (ceilDiv(T0.size[1], i22)); ++i75) {
    float T12[1];
    float T11[1];
    if ((((i75 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T11[0]
        = T4[((i75 * blockDim.x) + threadIdx.x)]
        * T10[0];
    }

    if ((((i75 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T12[0]
        = T11[0]
        * f4;
    }

    if ((((i75 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T13[(blockIdx.x * T13.stride[0]) + ((i75 * blockDim.x) + threadIdx.x)]
        = T12[0]
        + f5;
    }
  }
}
```

**Batch Norm Persistent**
```
__global__ void CUDAGeneratedKernel(Tensor<float, 2> T0, float f4, float f5, float f6, int64_t i7, int64_t i23, Tensor<float, 2> T13) {
  alignas(4) extern __shared__ char array[];
  unsigned offset = 0;

  void* shared_mem = array;
  offset += ((blockDim.x * blockDim.y * blockDim.z) * sizeof(float));

  offset = alignBufferSize(offset,4);
  float* T14 = reinterpret_cast<float*>(array + offset);
  offset += (((ceilDiv(T0.size[1], i23)) * i23) * sizeof(float));

  offset = alignBufferSize(offset,4);
  float* T4 = reinterpret_cast<float*>(array + offset);
  offset += (((ceilDiv(T0.size[1], i23)) * i23) * sizeof(float));

  float T10[1];
  float T9[1];
  float T8[1];
  float T7[1];

  float T6[1];
  T6[0] = float(0);

  float T16[1];
  T16[0] = float(0);

  float T3[1];
  float T2[1];

  float T1[1];
  T1[0] = float(0);

  float T15[1];
  T15[0] = float(0);

  // Read input into shared memory
  for(size_t i4 = 0; i4 < (ceilDiv(T0.size[1], i23)); ++i4) {
    if ((((i4 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T14[((i4 * blockDim.x) + threadIdx.x)]
         = T0[(blockIdx.x * T0.stride[0]) + (((i4 * i23) + threadIdx.x) * T0.stride[1])];
    }
  }
  __syncthreads();

  // Thread Independent Sum
  for(size_t i10 = 0; i10 < (ceilDiv(T0.size[1], i23)); ++i10) {
    if ((((i10 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T15[0]
        = T15[0]
        + T14[((i10 * blockDim.x) + threadIdx.x)];
    }
  }

  // Mean
  blockReduce<true, false, false>(
    T1[0],
    T15[0],
    [](float &a, float b) { a = a + b; },
    threadIdx,
    blockDim,
    static_cast<float*>(shared_mem),
    1,
    float(0));

  broadcast::blockBroadcast<true, false, false>(
    T2[0],
    T1[0],
    static_cast<float*>(shared_mem));

  T3[0]
    = T2[0]
    / i7;

  // Load input minus input_mean into shared memory
  for(size_t i31 = 0; i31 < (ceilDiv(T0.size[1], i23)); ++i31) {
    if ((((i31 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T4[((i31 * blockDim.x) + threadIdx.x)]
        = T14[((i31 * blockDim.x) + threadIdx.x)]
        - T3[0];
    }
  }
  __syncthreads();

  for(size_t i36 = 0; i36 < (ceilDiv(T0.size[1], i23)); ++i36) {
    float T5[1];
    // Square Difference
    if ((((i36 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T5[0]
        = T4[((i36 * blockDim.x) + threadIdx.x)]
        * T4[((i36 * blockDim.x) + threadIdx.x)];
    }
    // Thread Independent Sum
    if ((((i36 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T16[0]
        = T16[0]
        + T5[0];
    }
  }

  // Variance
  blockReduce<true, false, false>(
    T6[0],
    T16[0],
    [](float &a, float b) { a = a + b; },
    threadIdx,
    blockDim,
    static_cast<float*>(shared_mem),
    1,
    float(0));

  broadcast::blockBroadcast<true, false, false>(
    T7[0],
    T6[0],
    static_cast<float*>(shared_mem));

  T8[0]
    = T7[0]
    / i7;

  // Var + EPS
  T9[0]
    = T8[0]
    + f6;

  // 1 / Sqrt(Var + EPS)
  T10[0]
     = rsqrtf(T9[0]);

  for(size_t i75 = 0; i75 < (ceilDiv(T0.size[1], i23)); ++i75) {
    float T12[1];
    float T11[1];
    // Norm = (x - mean) / sqrt(var + eps)
    if ((((i75 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T11[0]
        = T4[((i75 * blockDim.x) + threadIdx.x)]
        * T10[0];
    }
    // Norm * Gamma
    if ((((i75 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T12[0]
        = T11[0]
        * f4;
    }
    // Norm * Gamma + Beta
    if ((((i75 * blockDim.x) + threadIdx.x) < T0.size[1])) {
      T13[(blockIdx.x * T13.stride[0]) + ((i75 * blockDim.x) + threadIdx.x)]
        = T12[0]
        + f5;
    }
  }
}
```
